demisto_sdk/commands/run_cmd/runner.py

```diff
@@ -312,3 +339,97 @@ def execute_command(self, command: str):
+                "",  # TODO
+            )
+        logger.debug(f"about={raw_response}")
+
+        if (
+            raw_response.get("productMode", "") == "xsoar"
+            and raw_response.get("deploymentMode", "") == "saas"
+        ):
+            return ServerType.XSOAR_SAAS
+        return ServerType.XSIAM
+
+    def create_incident_for_query(self):
+        """
+        Creates an incident if the server is XSOAR SaaS and returns the incident ID.
+        Otherwise, no incident is created, and None is returned.
+        """
+
+        # Check if the server is XSOAR 8
+        if self.server_type is not ServerType.XSOAR_SAAS:
+            self.is_incident_runtime_created = False
+            return
+
+        incident, status_code, _ = self.client.generic_request(
+            path="/incident",
+            method="POST",
+            content_type="application/json",
+            body={
+                "createInvestigation": True,
+                "name": "Demisto-sdk Run",
+                "owner": "",
+                "severity": 0,
+                "type": "Unclassified",
+                "playbook": "",
+            },
+            response_type=object,
+        )
+
+        if status_code != 201:
+            logger.critical(
+                "Failed to create a new incident for executing the command."
+            )
+            raise RuntimeError(
+                "Failed to create a new incident for executing the command."
+            )
+        logger.info(
+            f"Incident was successfully created for executing the command - ID: {incident['id']}."
+        )
+        self.is_incident_runtime_created = True
+        return incident["id"]
+
+    def delete_incident_if_needed(self):
+        if not self.is_incident_runtime_created:
+            return
+
+        body: dict[str, Any] = {}
+        if self.incident_post_run == IncidentStatus.OPEN:
+            logger.info(f"The incident {self.incident_id} remains open.")
+            return
+        elif self.incident_post_run == IncidentStatus.CLOSE:
+            path = "/incident/close"
+            body = {"CustomFields": {}, "id": self.incident_id}
+        elif self.incident_post_run == IncidentStatus.REMOVE:
+            path = "/incident/batchDelete"
+            body = {"ids": [self.incident_id], "all": False, "filter": {}}
+        else:
+            raise DemistoRunTimeError(
+                "The argument `incident_post_run` was not set correctly.\n"
+                "Allowed values are: `open`, `close`, `remove`."
+            )
+
+        _, status_code, _ = self.client.generic_request(
+            path=path,
+            method="POST",
+            content_type="application/json",
+            body=body,
+            response_type=object,
+        )
+
+        if status_code != 200:
+            logger.info(
+                f"Failed {self.incident_post_run.value}d the incident {self.incident_id}"
+            )
+        logger.info(
```

Comment:
Can we add the raw response from the `client.generic_request` in the log in case of a non-200 response? It could help debugging in the future.